DataFormats/PatCandidates/interface/PATObject.h

```diff
@@ -768,6 +770,24 @@ namespace pat {
   }
 
   template <class ObjectType>
+  void PATObject<ObjectType>::addUserDataObject_( const std::string & label, std::auto_ptr<pat::UserData> & data, bool overwrite ) 
```

Comment:
I think adding the new interface since it would allow us an incremental adoption path.